src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -3045,53 +3032,71 @@ void HierRTLMP::calculateMacroTilings(Cluster* cluster)
   debugPrint(logger_, MPL, "coarse_shaping", 2, "{}", line);
 }
 
-// Create pin blockage for Bundled IOs
-// Each pin blockage is a macro only blockage
 void HierRTLMP::setIOClustersBlockages()
 {
-  debugPrint(logger_,
-             MPL,
-             "coarse_shaping",
-             1,
-             "Creating the pin blockage for root cluster");
-
   if (!io_pad_map_.empty()) {
     return;
   }
 
-  const float root_lx = root_cluster_->getX();
-  const float root_ly = root_cluster_->getY();
-  const float root_ux = root_cluster_->getX() + root_cluster_->getWidth();
-  const float root_uy = root_cluster_->getY() + root_cluster_->getHeight();
+  IOSpans io_spans = computeIOSpans();
+  const float depth = computeIOBlockagesDepth(io_spans);
 
-  const std::vector<std::pair<float, float>> tilings
-      = root_cluster_->getMacroTilings();
-  const float ar = root_cluster_->getHeight() / root_cluster_->getWidth();
-  float max_height = std::sqrt(tilings[0].first * tilings[0].second * ar);
-  float max_width = max_height / ar;
-  max_width = ((root_ux - root_lx) - max_width);
-  max_height = ((root_uy - root_ly) - max_height);
+  const Rect root(root_cluster_->getX(),
```

Comment:
Cluster::getX returns a float - is this intended to be rounded to an int?  Likewise the rest.

Reply:
Here Rect is mpl2::Rect which is in micron, not odb::Rect

Reply:
Ok (though we really need to nuke this micron code)